src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/Assert.cs

```diff
@@ -384,18 +385,37 @@ public static void AssemblyContainsType(MSBuildResult result, string assemblyPat
             }
 
             assemblyPath = Path.Combine(result.Project.DirectoryPath, Path.Combine(assemblyPath));
+
+            var typeNames = GetDeclaredTypeNames(assemblyPath);
+            Assert.Contains(fullTypeName, typeNames);
+        }
+
+        public static void AssemblyDoesNotContainType(MSBuildResult result, string assemblyPath, string fullTypeName)
+        {
+            if (result == null)
+            {
+                throw new ArgumentNullException(nameof(result));
+            }
+
+            var typeNames = GetDeclaredTypeNames(assemblyPath);
+            Assert.DoesNotContain(fullTypeName, typeNames);
+        }
+
+        private static IEnumerable<string> GetDeclaredTypeNames(string assemblyPath)
+        {
+            IEnumerable<string> typeNames;
             using (var file = File.OpenRead(assemblyPath))
             {
                 var peReader = new PEReader(file);
                 var metadataReader = peReader.GetMetadataReader();
-                var typeNames = metadataReader.TypeDefinitions.Where(t => !t.IsNil).Select(t =>
+                typeNames = metadataReader.TypeDefinitions.Where(t => !t.IsNil).Select(t =>
                 {
                     var type = metadataReader.GetTypeDefinition(t);
                     return metadataReader.GetString(type.Namespace) + "." + metadataReader.GetString(type.Name);
                 });
-
-                Assert.Contains(fullTypeName, typeNames);
             }
+
+            return typeNames;
```

Comment:
Why not keep the return inside the using and do a `.ToArray`. This is the obtuse.